src/main/java/homework_2/pyramid_printer/PyramidPrinter.java

```diff
@@ -0,0 +1,52 @@
+package homework_2.pyramid_printer;
+
+import homework_2.pyramid_printer.exception.PyramidPrinterException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import static homework_2.pyramid_printer.utils.MessageType.ERROR_MESSAGE;
+import static homework_2.pyramid_printer.utils.MessageType.ERROR_WRONG_FORMAT_MESSAGE;
+import static homework_2.pyramid_printer.utils.MessageType.INFO_MESSAGE;
+import static homework_2.pyramid_printer.utils.Utils.printMessage;
+import static java.lang.System.lineSeparator;
+
+public class PyramidPrinter {
+    private static final String LETTER = "x";
+
+    public void start() {
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
+            printMessage(INFO_MESSAGE.getMessage());
+            String line = reader.readLine();
+            int rows = getNumber(line);
+            generatePyramid(rows);
+        } catch (IOException e) {
+            printMessage(ERROR_MESSAGE.getMessage());
+        } catch (PyramidPrinterException e) {
+            printMessage(e.getMessage());
+        }
+    }
+
+    private int getNumber(String text) throws PyramidPrinterException {
+        int number;
+        try {
+            number = Integer.parseInt(text);
+            if (number < 1) {
+                throw new PyramidPrinterException(ERROR_WRONG_FORMAT_MESSAGE);
+            }
+        } catch (NumberFormatException e) {
+            throw new PyramidPrinterException(ERROR_WRONG_FORMAT_MESSAGE);
```

Comment:
opt: throwing exceptions is heavyweight and if possible better avoided

Reply:
I'd tried to do something with it but it is really hard to get rid of throwing exceptions in those methods so I decided to leave it like this for now but I will keep your note in mind and try to avoid throwing exceptions in the future